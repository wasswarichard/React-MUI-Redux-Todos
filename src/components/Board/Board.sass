$fontFamily: 'Lato', helvetica, Arial, Sans-serif

// Backgrounds
$bodyBg: #f2f2f2
$columnBg: #f5f5f5
$taskCardBorder: #E0E1E2
$backgroundColor: #CDCED0

// Colors
$black: #333


body
  background: white

.board
  display: flex
  flex-wrap: nowrap
  justify-content: space-around
  //padding: 15px
  height: 10px
  //max-height: 100px
  //min-width: 1000px
  width: 80%
  margin: 5px auto 50px auto
  //margin: 50px auto
  //align-content: stretch


.TodoList
  font-family: $fontFamily
  margin: 5px auto 5px auto
  .boardColumn-taskList
    background: $columnBg
    margin-left: 30px

    //padding: 20px 10px
    //min-height: 600px
    height: 300px
    //max-height: 600px
    overflow-y: auto


.boardColumn
  font-family: $fontFamily
  flex-basis: 300px
  flex-grow: 1
  margin: 0 5px

  .boardColumn-header
    padding: 5px
    margin-bottom: 10px
    border-bottom: 5px solid $columnBg
  .boardColumn-title
    text-transform: capitalize
    color: $black
    font-size: 14px
    line-height: 1.5
    font-weight: normal
    margin: 0 5px 0 0
    word-wrap: break-word
  .boardColumn-taskList
    background: $columnBg
    padding: 20px 10px
    //min-height: 600px
    height: 300px
    //max-height: 600px
    overflow-y: auto

.taskCard
  font-family: $fontFamily
  border: 2px solid $taskCardBorder
  border-radius: 15px
  padding: 15px 15px 10px 15px
  background: white
  position: relative
  margin-bottom: 5px
  box-shadow: 0px 3px 4px 0px rgba(0, 0, 0, 0.06)

  .taskCard-profile
    margin-left: -5px
    width: 95%

  .taskCard-image
    border-radius: 30%
    max-width: 25%

  .taskCard-info
    font-size: 11px
    margin-left: 55px
    margin-top: -55px
    align-items: center

  .label-value
    color: #3c8bd0

