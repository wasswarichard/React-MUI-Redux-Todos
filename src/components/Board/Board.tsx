import React from 'react';
import "./Board.sass"
import BoardColumn from "../BoardColumn/BoardColumn";
import {status} from "../../constants/status";
import {ITODO} from "../../types/types";

const Board = ({todos, setStatusChange}: {todos: ITODO[] , setStatusChange: any}) => {
    return (
        <div className="board">
            <BoardColumn title = "TODO"  todos = { todos?.filter( (todo: ITODO )=> {return todo.status === status[0]})} setStatusChange={setStatusChange} />
            <BoardColumn title = "IN_PROGRESS" todos = {todos?.filter( (todo: ITODO) => {return todo.status === status[1]})} setStatusChange={setStatusChange}/>
            <BoardColumn title = "DONE" todos = { todos?.filter( (todo: ITODO) => {return todo.status === status[2]})} setStatusChange={setStatusChange}/>
        </div>
    );
}
export default Board;