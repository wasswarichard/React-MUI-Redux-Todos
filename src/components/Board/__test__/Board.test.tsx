import React from 'react';
import Board from '../Board';
import Enzyme, { shallow } from 'enzyme';
import { cleanup } from '@testing-library/react';
import Adapter from 'enzyme-adapter-react-16';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
Enzyme.configure({ adapter: new Adapter() });

describe('<Board>', () => {
  afterEach(cleanup);
  const todoElements = [
    { userId: 1, id: 1, title: 'DONE', completed: true, status: 'DONE' },
    { userId: 2, id: 2, title: 'DONE', completed: true, status: 'DONE' },
  ];
  const statusChange = () => {};
  const mockStore = configureStore();
  let store;

  it('renders BoardColumn Component', () => {
    shallow(<Board todos={todoElements} setStatusChange={statusChange} />);
  });

  it('should match snapshot', function () {
    store = mockStore(todoElements);
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Board todos={todoElements} setStatusChange={statusChange} />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
