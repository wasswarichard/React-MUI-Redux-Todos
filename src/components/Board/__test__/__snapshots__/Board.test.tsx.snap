// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board> should match snapshot 1`] = `
<div
  className="board"
>
  <div
    className="boardColumn"
  >
    <header
      className="boardColumn-header"
    >
      <h1
        className="boardColumn-title"
      >
        TODO
      </h1>
    </header>
    <div
      className="boardColumn-taskList"
    />
  </div>
  <div
    className="boardColumn"
  >
    <header
      className="boardColumn-header"
    >
      <h1
        className="boardColumn-title"
      >
        IN_PROGRESS
      </h1>
    </header>
    <div
      className="boardColumn-taskList"
    />
  </div>
  <div
    className="boardColumn"
  >
    <header
      className="boardColumn-header"
    >
      <h1
        className="boardColumn-title"
      >
        DONE
      </h1>
    </header>
    <div
      className="boardColumn-taskList"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root MuiButtonBase-root-card css-1hgxlf6-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiButtonBase-root-avatar"
        >
          <img
            alt="profile picture"
            className="MuiAvatar-img"
            src="anonymous.jpg"
          />
        </div>
        <div
          className="MuiButtonBase-root-content"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            1
            . 
            <span>
              DONE
            </span>
          </p>
        </div>
        <div>
          <hr />
          <div
            className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <button
              className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Decline
            </button>
            <button
              className="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-kmxb0e-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Approve
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root MuiButtonBase-root-card css-1hgxlf6-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiButtonBase-root-avatar"
        >
          <img
            alt="profile picture"
            className="MuiAvatar-img"
            src="anonymous.jpg"
          />
        </div>
        <div
          className="MuiButtonBase-root-content"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            2
            . 
            <span>
              DONE
            </span>
          </p>
        </div>
        <div>
          <hr />
          <div
            className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <button
              className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Decline
            </button>
            <button
              className="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-kmxb0e-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Approve
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<BoardColumn> should match snapshot 1`] = `
<div
  className="board"
>
  <div
    className="boardColumn"
  >
    <header
      className="boardColumn-header"
    >
      <h1
        className="boardColumn-title"
      >
        TODO
      </h1>
    </header>
    <div
      className="boardColumn-taskList"
    />
  </div>
  <div
    className="boardColumn"
  >
    <header
      className="boardColumn-header"
    >
      <h1
        className="boardColumn-title"
      >
        IN_PROGRESS
      </h1>
    </header>
    <div
      className="boardColumn-taskList"
    />
  </div>
  <div
    className="boardColumn"
  >
    <header
      className="boardColumn-header"
    >
      <h1
        className="boardColumn-title"
      >
        DONE
      </h1>
    </header>
    <div
      className="boardColumn-taskList"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root MuiButtonBase-root-card css-1hgxlf6-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiButtonBase-root-avatar"
        >
          <img
            alt="profile picture"
            className="MuiAvatar-img"
            src="anonymous.jpg"
          />
        </div>
        <div
          className="MuiButtonBase-root-content"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            1
            . 
            <span>
              DONE
            </span>
          </p>
        </div>
        <div>
          <hr />
          <div
            className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <button
              className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Decline
            </button>
            <button
              className="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-kmxb0e-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Approve
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root MuiButtonBase-root-card css-1hgxlf6-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiButtonBase-root-avatar"
        >
          <img
            alt="profile picture"
            className="MuiAvatar-img"
            src="anonymous.jpg"
          />
        </div>
        <div
          className="MuiButtonBase-root-content"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            2
            . 
            <span>
              DONE
            </span>
          </p>
        </div>
        <div>
          <hr />
          <div
            className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <button
              className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Decline
            </button>
            <button
              className="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-kmxb0e-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Approve
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
